data/index.js

```diff
@@ -139,7 +155,18 @@ export function select( reducerKey ) {
 		return select( reducerKey )[ selectorKey ]( ...args );
 	}
 
-	return selectors[ reducerKey ];
+	const createResolver = ( selector, key ) => ( ...args ) => {
+		const result = selector( ...args );
+		const resolver = resolvers[ reducerKey ] && resolvers[ reducerKey ][ key ];
+		if ( ! resolver ) {
+			return result;
+		}
+		resolver.fulfill( ...args );
+
+		return result;
+	};
+
+	return mapValues( selectors[ reducerKey ], createResolver );
```

Comment:
This seems particularly heavy for an operation which we're assuming to occur very frequently (`select`). Could the selectors be augmented once as part of `registerResolvers` instead?